### PR TITLE
LPFG-1130: Change my email

### PR DESCRIPTION
**Pull request by robleney-kainos**

### CODE REVIEW

File: src/main/java/uk/gov/cshr/domain/EmailUpdate.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import java.util.UUID;
+
+@Data
+@Entity
+public class EmailUpdate {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private long id;
+
+    private String code = UUID.randomUUID().toString();
```
Comment (elxsy) [inline src/main/java/uk/gov/cshr/domain/EmailUpdate.java]: @m24murray random codes should always be url safe random strings, not guessible UUIDs.
```suggestion
    private String code = RandomStringUtils.random(40, true, true);
```

File: src/main/resources/db/migration/h2/V1__init.sql [hunk 2]
```diff
@@ -75,3 +75,13 @@ CREATE TABLE `token` (
   `user_name` varchar(255) DEFAULT NULL,
   PRIMARY KEY (`id`)
 );
+
+CREATE TABLE `email_update` (
+  `id` int(10) unsigned NOT NULL AUTO_INCREMENT,
+  `code` char(36) NOT NULL,
```
Comment (elxsy) [inline src/main/resources/db/migration/h2/V1__init.sql]: @m24murray we want code to be 40 characters long and unique `index`

File: src/main/resources/db/migration/mysql/V1.2.5__add-email-update.sql [hunk 3]
```diff
@@ -0,0 +1,10 @@
+CREATE TABLE `email_update` (
+  `id` int(10) unsigned NOT NULL AUTO_INCREMENT,
+  `code` char(36) NOT NULL,
```
Comment (elxsy) [inline src/main/resources/db/migration/mysql/V1.2.5__add-email-update.sql]: @m24murray same as above

**Comment by Peter-D-Taylor** [comment]
Also requires Oauth2 for the notification service to work

**Pull request closed without merging**